core/src/subject/SubjectEntity.ts

```diff
@@ -49,75 +231,192 @@ export class SubjectEntity {
+    // + offset
+    // + include
+
+    const subQueries = [
+      buildSourceQuery(query?.source),
+      buildAuthorAndTimestampQuery(query?.properties),
+      buildPropertiesQuery(query?.properties),
+      buildCollectionsQuery(query?.collections),
+      buildWhereQuery(query?.where),
+    ];
+
+    const fullQuery = `
+      findall([Base, Properties, Collections, Timestamp, Author], (
+        subject_class("${await this.getClassName(perspective)}", SubjectClass),
+        instance(SubjectClass, Base),
+        ${subQueries.filter((q) => q).join(", ")}
+      ), AllInstances)
+    `;
+
+    if (query?.where) console.log("whereQuery!!!", buildWhereQuery(query?.where));
+
+    const result = await perspective.infer(fullQuery);
+    if (!result?.[0]?.AllInstances) return [];
+
+    // Map results to instances
+    const requestedAttribtes = [...(query?.properties || []), ...(query?.collections || [])];
+    const allInstances = await Promise.all(
+      result[0].AllInstances.map(async ([Base, Properties, Collections, Timestamp, Author]) => {
+        const instance = new this(perspective, Base) as any;
+        // Remove unrequested attributes from instance
+        if (requestedAttribtes.length) {
+          Object.keys(instance).forEach((key) => {
+            if (!requestedAttribtes.includes(key)) delete instance[key];
+          });
+        }
+        // Collect values to assign to instance
+        const values = [...Properties, ...Collections];
+        if (!query?.properties || query.properties.includes("timestamp")) values.push(["timestamp", Timestamp]);
+        if (!query?.properties || query.properties.includes("author")) values.push(["author", Author]);
+        await SubjectEntity.assignValuesToInstance(perspective, instance, values);
+
+        return instance;
+      })
+    );
+
+    return allInstances;
   }
 
   private async setProperty(key: string, value: any) {
-    const setters = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), property_setter(C, "${key}", Setter)`)
+    const setters = await this.#perspective.infer(
+      `subject_class("${this.#subjectClassName}", C), property_setter(C, "${key}", Setter)`
+    );
     if (setters && setters.length > 0) {
-      const actions = eval(setters[0].Setter)
-      const resolveLanguageResults = await this.#perspective.infer(`subject_class("${this.#subjectClass}", C), property_resolve_language(C, "${key}", Language)`)
-      let resolveLanguage
+      const actions = eval(setters[0].Setter);
+      const resolveLanguageResults = await this.#perspective.infer(
+        `subject_class("${this.#subjectClassName}", C), property_resolve_language(C, "${key}", Language)`
+      );
+      let resolveLanguage;
       if (resolveLanguageResults && resolveLanguageResults.length > 0) {
-        resolveLanguage = resolveLanguageResults[0].Language
+        resolveLanguage = resolveLanguageResults[0].Language;
       }
 
       if (resolveLanguage) {
```

Comment:
Hm, weird, so we do have the code that would create the literal URLs from values when resolve language is set. Should investigate if and/or when that is failing...